hooks/browser_integration.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that provides utilities used by Toolkit's browser integration.
+"""
+
+import os
+import hashlib
+import json
+import fnmatch
+import datetime
+
+import sgtk
+
+class BrowserIntegration(sgtk.Hook):
+    def filter_engine_commands(self, commands):
+        """
+        Filters out engine commands that are not desired in the web app menus.
```

Comment:
I'd like to make it customizable. However, I like the idea of presenting it at process_commands.